src/java/voldemort/server/protocol/admin/AsyncOperation.java

```diff
@@ -29,13 +29,17 @@ public void markComplete() {
 
     public void run() {
         updateStatus("Started " + getStatus());
+        String previousThreadName = Thread.currentThread().getName();
+        Thread.currentThread().setName(previousThreadName + "; AsyncOperation ID " + status.getId() + ": " + status.getDescription());
```

Comment:
What does description contain ? If it is very long, probably you can replace it with System.currentTimeMillis(). If it is short and concise you can ignore this.